plugins/woocommerce/src/Blocks/BlockTypes/MiniCartTitleItemsCounterBlock.php

```diff
@@ -11,4 +13,68 @@ class MiniCartTitleItemsCounterBlock extends AbstractInnerBlock {
 	 * @var string
 	 */
 	protected $block_name = 'mini-cart-title-items-counter-block';
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content    Block content.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		if ( Features::is_enabled( 'experimental-iapi-mini-cart' ) ) {
+			return $this->render_experimental_iapi_title_label_block();
+		}
+		return $content;
+	}
+
+	/**
+	 * Render the interactivity API powered experimental title block.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	protected function render_experimental_iapi_title_label_block() {
+		$cart            = $this->get_cart_instance();
+		$cart_item_count = $cart ? $cart->get_cart_contents_count() : 0;
+		// translators: %d number of items in the cart.
+		$cart_item_text = _n( '(%d item)', '(%d items)', $cart_item_count, 'woocommerce' );
+
+		// translators: item is an item in cart.
+		$singular = __( '(%d item)', 'woocommerce' );
+		// translators: items is items in a cart.
+		$plural = __( '(%d items)', 'woocommerce' );
+
+		$context = array(
+			'singularItemsText' => $singular,
+			'pluralItemsText'   => $plural,
+		);
+
+		ob_start();
+		?>
+		<?php // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+		<span data-wp-text="state.itemsInCartText" data-wp-interactive="woocommerce/mini-cart-title-items-counter-block" <?php echo wp_interactivity_data_wp_context( $context ); ?> class="wp-block-woocommerce-mini-cart-title-items-counter-block">
+					<?php
+						// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+						printf( $cart_item_text, $cart_item_count );
```

Comment:
This should be populated by the iAPI by providing `state.itemsInCartText` in the server.

Reply:
Good point, I really got caught up in this pattern for some reason 😅 

Resolved in https://github.com/woocommerce/woocommerce/pull/58265/commits/b1d5a2ea53ba04144cc91b6e59d5d6a824b3a5d7